StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/SavedPaymentMethodFormFactory.swift

```diff
@@ -0,0 +1,148 @@
+}
+
+class SavedPaymentMethodFormFactory {
+    let viewModel: UpdatePaymentMethodViewModel
+    weak var delegate: SavedPaymentMethodFormFactoryDelegate?
+
+    init(viewModel: UpdatePaymentMethodViewModel) {
+        self.viewModel = viewModel
+    }
+
+    func makePaymentMethodForm() -> UIView {
+        switch viewModel.paymentMethod.type {
+        case .card:
+            return cardSection.view
+        case .USBankAccount:
+            return usBankAccountSection
+        case .SEPADebit:
+            return sepaDebitSection
+        default:
+            fatalError("Cannot make payment method form for payment method type \(viewModel.paymentMethod.type).")
+        }
+    }
+
+    private lazy var cardBrandDropDown: DropdownFieldElement? = {
+        guard viewModel.paymentMethod.isCoBrandedCard else { return nil }
+        let cardBrands = viewModel.paymentMethod.card?.networks?.available.map({ STPCard.brand(from: $0) }).filter { viewModel.cardBrandFilter.isAccepted(cardBrand: $0) } ?? []
+                    let cardBrandDropDown = DropdownFieldElement.makeCardBrandDropdown(cardBrands: Set<STPCardBrand>(cardBrands),
+                                                                                       theme: viewModel.appearance.asElementsTheme,
+                                                                                       includePlaceholder: false) { [self] in
+                        let selectedCardBrand = viewModel.selectedCardBrand ?? .unknown
+                        let params = ["selected_card_brand": STPCardBrandUtilities.apiValue(from: selectedCardBrand), "cbc_event_source": "edit"]
+                        STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: viewModel.hostedSurface.analyticEvent(for: .openCardBrandDropdown),
+                                                                             params: params)
+                    } didTapClose: { [self] in
+                        let selectedCardBrand = viewModel.selectedCardBrand ?? .unknown
+                        STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: viewModel.hostedSurface.analyticEvent(for: .closeCardBrandDropDown),
+                                                                             params: ["selected_card_brand": STPCardBrandUtilities.apiValue(from: selectedCardBrand)])
+                    }
+
+                    // pre-select current card brand
+                    if let currentCardBrand = viewModel.paymentMethod.card?.preferredDisplayBrand,
+                       let indexToSelect = cardBrandDropDown.items.firstIndex(where: { $0.rawData == STPCardBrandUtilities.apiValue(from: currentCardBrand) }) {
+                        cardBrandDropDown.select(index: indexToSelect, shouldAutoAdvance: false)
+                        viewModel.selectedCardBrand = currentCardBrand
+                    }
+                    return cardBrandDropDown
+    }()
+
+    private lazy var panElement: TextFieldElement = {
```

Comment:
Probably not in this PR but another one, it might be a good idea to move these all to their own file, it reads kinda hard right now with all the lazy variables (most of which aren't used for certain payment methods). See PaymentSheetFormFactory for what that may look like.